plugins/optimization-detective/class-od-element.php

```diff
@@ -0,0 +1,209 @@
+	}
+
+	/**
+	 * Gets property value for an arbitrary key.
+	 *
+	 * This is particularly useful in conjunction with the `od_url_metric_schema_element_item_additional_properties` filter.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $key Property.
+	 * @return mixed|null The property value, or null if not set.
+	 */
+	public function get( string $key ) {
+		return $this->data[ $key ] ?? null;
+	}
+
+	/**
+	 * Determines whether element was detected as LCP.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return bool Whether LCP.
+	 */
+	public function is_lcp(): bool {
+		return $this->data['isLCP'];
+	}
+
+	/**
+	 * Determines whether element was detected as an LCP candidate.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return bool Whether LCP candidate.
+	 */
+	public function is_lcp_candidate(): bool {
+		return $this->data['isLCPCandidate'];
+	}
+
+	/**
+	 * Gets XPath for element.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return non-empty-string XPath.
+	 */
+	public function get_xpath(): string {
+		return $this->data['xpath'];
+	}
+
+	/**
+	 * Gets intersectionRatio for element.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return float Intersection ratio.
+	 */
+	public function get_intersection_ratio(): float {
+		return $this->data['intersectionRatio'];
+	}
+
+	/**
+	 * Gets intersectionRect for element.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @phpstan-return DOMRect
+	 *
+	 * @return array Intersection rect.
+	 */
+	public function get_intersection_rect(): array {
+		return $this->data['intersectionRect'];
+	}
+
+	/**
+	 * Gets boundingClientRect for element.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @phpstan-return DOMRect
+	 *
+	 * @return array Bounding client rect.
+	 */
+	public function get_bounding_client_rect(): array {
+		return $this->data['boundingClientRect'];
+	}
+
+	/**
+	 * Checks whether an offset exists.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param mixed $offset Key.
+	 * @return bool Whether the offset exists.
+	 */
+	public function offsetExists( $offset ): bool {
+		return isset( $this->data[ $offset ] );
+	}
+
+	/**
+	 * Retrieves an offset.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @template T of key-of<ElementData>
+	 * @phpstan-param T $offset
+	 * @phpstan-return ElementData[T]|null
+	 * @todo This should account for additional undefined keys which can be supplied by extending the element schema. May depend on <https://github.com/phpstan/phpstan/issues/8438>.
+	 *
+	 * @param mixed $offset Key.
+	 * @return mixed May return any value from ElementData including possible extensions.
+	 */
+	public function offsetGet( $offset ) {
+		return $this->data[ $offset ] ?? null;
+	}
+
+	/**
+	 * Sets an offset.
+	 *
+	 * This is disallowed. Attempting to set a property will throw an error.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param mixed $offset Key.
+	 * @param mixed $value  Value.
+	 *
+	 * @throws Exception When attempting to set a property.
+	 */
+	public function offsetSet( $offset, $value ): void {
+		throw new Exception( 'Offsets may not be set.' );
```

Comment:
Maybe more clear as something like this?
```suggestion
		throw new Exception( 'Element data may not be set.' );
```